### PR TITLE
test: Code Coverage: Receipt Preview Component

### PR DESCRIPTION
**Pull request by jayfyle**
### Clickup: https://app.clickup.com/t/85zrxn9rc

# Coverage at 95%

receipt_preview_95

### CODE REVIEW

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts [hunk 1]
```diff
@@ -1,26 +1,417 @@
...
+
+  it('ionSlideDidChange(): should detect slide change and update active index', () => {
+    component.ionSlideDidChange();
+    expect(component.activeIndex).toEqual(0);
+  });
+
+  it('addMore(): should add more receipts', async () => {
+    matBottomSheet.open.and.returnValue({
+      afterDismissed: () =>
+        of({
```
Comment (Julias0) [inline src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:398]: same as [this](https://github.com/fylein/fyle-mobile-app/pull/1953/files#r1175321691)

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts [hunk 2]
```diff
@@ -1,26 +1,417 @@
...
+        cssClass: 'pop-up-in-center',
+      });
+    });
+
+    it('should retake images if there are no existing images to delete', async () => {
+      component.base64ImagesWithSource = [];
+      spyOn(component, 'retake').and.returnValue(null);
+      fixture.detectChanges();
+      popoverController.create.and.returnValue(
+        new Promise((resolve) => {
```
Comment (Julias0) [inline src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:326]: remove nesting using promise.resolve

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts [hunk 3]
```diff
@@ -1,26 +1,417 @@
...
+      continueCaptureReceipt: true,
+    });
+  });
+
+  describe('deleteReceipt():', () => {
+    it('should delete receipt', async () => {
+      component.base64ImagesWithSource = images;
+      fixture.detectChanges();
+      popoverController.create.and.returnValue(
+        new Promise((resolve) => {
```
Comment (Julias0) [inline src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:286]: remove nesting with promise.resolve

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts [hunk 4]
```diff
@@ -1,26 +1,417 @@
...
+          },
+        },
+        cssClass: 'pop-up-in-center',
+      });
+    });
+
+    it('should close modal and retake image if discard if no previous image exist', async () => {
+      component.base64ImagesWithSource = [];
+      popoverController.create.and.returnValue(
+        new Promise((resolve) => {
```
Comment (Julias0) [inline src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:216]: remove nesting with promise.resolve

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts [hunk 5]
```diff
@@ -1,26 +1,417 @@
...
+      base64ImagesWithSource: component.base64ImagesWithSource,
+    });
+  });
+
+  describe('closeModal():', () => {
+    it('should close modal and retake image if discard is selected on existing image', async () => {
+      spyOn(component, 'retake').and.returnValue(null);
+      component.base64ImagesWithSource = images;
+      popoverController.create.and.returnValue(
+        new Promise((resolve) => {
```
Comment (Julias0) [inline src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:174]: remove nesting with promise.resolve

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts [hunk 6]
```diff
@@ -1,26 +1,417 @@
...
+    fixture.detectChanges();
+  }));
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('openCropReceiptModal(): should open the crop receipt modal', async () => {
+    modalController.create.and.returnValue(
+      new Promise((resolve) => {
```
Comment (Julias0) [inline src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:119]: remove nesting with promise.resolve

**Review by Julias0 — CHANGES_REQUESTED** [review]
added few comments

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts (hunk 1 above)
Comment (jayfyle) [reply src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:398]: This returns a reference to material component ref, and the afterDismissed property returns an observable

File: src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts (hunk 6 above)
Comment (jayfyle) [reply src/app/shared/components/capture-receipt/receipt-preview/receipt-preview.component.spec.ts:119]: Refactoring the outer promise fails the spec and does not recognise the spy objects, so I am restructuring the inner promises

**Review by Julias0 — APPROVED** [review]

**Pull request merged**